test/interfaces/flags.test-types.ts

```diff
@@ -45,85 +100,242 @@ class MyCommand extends BaseCommand {
+    })(),
+    'option#defs:default': Flags.option({
+      options,
+      default: async _ctx => 'foo',
+    })(),
+    'option#defs:multiple': Flags.option({
+      options,
+      multiple: true,
+    })(),
+    'option#defs:multiple,required': Flags.option({
+      options,
+      multiple: true,
+      required: true,
+    })(),
+    'option#defs,multiple,default': Flags.option({
+      options,
+      multiple: true,
+      default: async _ctx => ['foo'],
+    })(),
+
+    'option#defs:multiple;opts:default': Flags.option({
+      options,
+      multiple: true,
+    })({
+      default: ['foo'],
+    }),
+
+    'option#defs:multiple;opts:default-callback': Flags.option({
+      options,
+      multiple: true,
+    })({
+      default: async _ctx => ['foo'],
+    }),
+
+    'custom#defs:multiple;opts:default-callback': Flags.custom({
+      options,
+      multiple: true,
+    })({
+      default: async _ctx => ['foo'],
+    }),
+
+    'custom#defs:multiple,parse': Flags.custom({
+      multiple: true,
+      parse: async (input, _ctx, _opts) => input,
+    })(),
+
+    'option#defs:multiple,prase': Flags.option({
+      options,
+      multiple: true,
+      parse: async (input, _ctx, _opts) => input as typeof options[number],
+    })(),
+
+    'custom#defs:multiple=true;opts:multiple=false': Flags.custom({
+      multiple: true,
+    })({
+      multiple: false,
+    }),
+    'custom#defs:multiple=false;opts:multiple=true': Flags.custom({
+      multiple: false,
+    })({
+      multiple: true,
+    }),
+    'custom#defs:required=true;opts:required=false': Flags.custom({
+      required: true,
+    })({
+      required: false,
+    }),
+    'custom#defs:required=false;opts:required=true': Flags.custom({
+      required: false,
+    })({
+      required: true,
+    }),
+    'custom#defs:multiple=true;opts:multiple=false,required=true': Flags.custom({
+      multiple: true,
+    })({
+      multiple: false,
+      required: true,
+    }),
+    'custom#defs:required=true;opts:multiple=true,required=false': Flags.custom({
+      required: true,
+    })({
+      multiple: true,
+      required: false,
+    }),
+    'custom#defs:required=false;opts:multiple=true,required=true': Flags.custom({
+      required: false,
+    })({
+      multiple: true,
+      required: true,
+    }),
+
+    'custom#defs:multiple=true,required=true;opts:multiple=false,required=false': Flags.custom({
+      multiple: true,
+      required: true,
+    })({
+      multiple: false,
+      required: false,
+    }),
+
+    'custom#defs:multiple=false,required=false;opts:multiple=true,required=true': Flags.custom({
+      multiple: false,
+      required: false,
+    })({
+      multiple: true,
+      required: true,
+    }),
+
+    'custom#defs:multiple=true;opts:multiple=false,default': Flags.custom({
+      multiple: true,
+    })({
+      multiple: false,
+      // TODO: THIS IS A BUG. It should enforce a single value instead of allowing a single value or an array
```

Comment:
did you mean to fix it for this major version?

Reply:
@mshanemc I'm probably not going to fix it in this PR (if at all). It's a very odd use case... someone defines a custom flag with `multiple: true` and then explicitly sets `multiple: false` when using the flag AND sets a default. Given my experience updating the flag types so far, I'd say it's likely a 2 day effort to fix this one edge case